diff --git a/stagit.c b/stagit.c
index 98035e9..33479f8 100644
--- a/stagit.c
+++ b/stagit.c
@@ -886,7 +886,7 @@ writeatom(FILE *fp, int all)
 }
 
 size_t
-writeblob(git_object *obj, const char *fpath, const char *filename, size_t filesize)
+writeblob(git_object *obj, const char *fpath, const char *filename, size_t filesize, const char *entrypath)
 {
 	char tmp[PATH_MAX] = "", *d;
 	const char *p;
@@ -908,10 +908,10 @@ writeblob(git_object *obj, const char *fpath, const char *filename, size_t files
 
 	fp = efopen(fpath, "w");
 	writeheader(fp, filename);
-	fputs("<p> ", fp);
+	fprintf(fp, "<a href=/raw/%s/%s> ", name, entrypath);
 	xmlencode(fp, filename, strlen(filename));
 	fprintf(fp, " (%zuB)", filesize);
-	fputs("</p><hr/>", fp);
+	fputs("</a><hr/>", fp);
 
 	if (git_blob_is_binary((git_blob *)obj)) {
 		fputs("<p>Binary file.</p>\n", fp);
@@ -1010,7 +1010,7 @@ writefilestree(FILE *fp, git_tree *tree, const char *path)
 			}
 
 			filesize = git_blob_rawsize((git_blob *)obj);
-			lc = writeblob(obj, filepath, entryname, filesize);
+			lc = writeblob(obj, filepath, entryname, filesize, entrypath);
 
 			fputs("<tr><td>", fp);
 			fputs(filemode(git_tree_entry_filemode(entry)), fp);
